pools-resolver/src/Resolver/Models/Types.hs

```diff
@@ -0,0 +1,20 @@
+module Resolver.Models.Types where
+
+import ErgoDex.Amm.Pool
+import Cardano.Models
+import Explorer.Types
+import Data.Aeson        (FromJSON(..), ToJSON(..))
+import GHC.Generics      (Generic)
+
+data OnChainIndexed a = OnChainIndexed { entity :: a, lastConfirmedBoxGix :: Gix }
+  deriving (Show, Generic, FromJSON, ToJSON)
+
+type PredictedPool = OnChainIndexed (FullTxOut, Pool)
+
+type ConfirmedPool = OnChainIndexed (FullTxOut, Pool)
```

Comment:
Try to avoid redundant generalization.

```suggestion
data PredictedPool = PredictedPool
  { predictedPool :: Pool
  , predictedOutput :: FullTxOut
  , lastConfirmedOutputGix :: Gix
  }

data ConfirmedPool = ConfirmedPool
  { confirmedPool :: Pool
  , confirmedOutput :: FullTxOut
  , lastConfirmedOutputGix :: Gix
  }
```
Here you'll have to use GADT in those points where you have to work with both types. 

or simpler version:

```suggestion
data LedgerPool = PredictedPool
  { pool :: Pool
  , output :: FullTxOut
  , lastConfirmedOutputGix :: Gix
  }
  
newtype PredictedPool = PredictedPool LedgerPool

newtype ConfirmedPool = ConfirmedPool LedgerPool
```